### PR TITLE
.Net4.x support and parallel test run fixes

### LINKED ISSUE #787: PR #781 broke the functional tests for dotnet frameworks (e.g.  net48) support
Error messages:
.../minio/minio-dotnet/Minio.Functional.Tests/FunctionalTest.cs(340,14): error CS0117: 'Parallel' does not contain a definition for 'ForEachAsync' [.../minio/minio-dotnet/Minio.Functional.Tests/Minio.Functional.Tests.csproj]
.../minio/minio-dotnet/Minio.Functional.Tests/Program.cs(252,14): error CS0117: 'Parallel' does not contain a definition for 'ForEachAsync' [.../minio/minio-dotnet/Minio.Functional.Tests/Minio.Functional.Tests.csproj]

### PR DESCRIPTION
**Pull request by ebozduman**
Fixes #787

Replaces the offending `Parallel.ForEachAsync()` function with its local implementation and fixed 2 other net4.x support problems I noticed.

I've tested and verified the fix with net6.0, net462 and net48.

### CODE REVIEW

**Comment by ebozduman** [comment]
@martijn00, @harshavardhana 

Could you review?

File: Minio/Helper/Utils.cs [hunk 1]
```diff
@@ -196,6 +197,20 @@ internal static bool IsSupersetOf(IList<string> l1, IList<string> l2)
         return !l2.Except(l1).Any();
     }
 
+    public static Task ForEachAsync<T>(this IEnumerable<T> source, int dop, Func<T, Task> body)
+    {
+        return Task.WhenAll(
```
Comment (martijn00) [inline Minio/Helper/Utils.cs:202]: Add a `#IF NET6` here and use my method when NET6 or higher is available. It is much more efficient.

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 2]
```diff
@@ -332,13 +332,9 @@ internal static async Task RunCoreTests(MinioClient minioClient)
...
         };
 
-        ParallelOptions parallelOptions = new()
-        {
-            MaxDegreeOfParallelism = 4
-        };
-        await Parallel
-            .ForEachAsync(coreTestsTasks, parallelOptions,
-                async (task, _) => { await task.ConfigureAwait(false); }).ConfigureAwait(false);
+        var MaxNoOfParallelProcesses = 4;
```
Comment (martijn00) [inline Minio.Functional.Tests/FunctionalTest.cs:335]: No need to put it in a local var, just add it in the util func and remove this from the method

File: Minio/Helper/Utils.cs [hunk 3]
```diff
@@ -196,6 +197,20 @@ internal static bool IsSupersetOf(IList<string> l1, IList<string> l2)
         return !l2.Except(l1).Any();
     }
 
+    public static Task ForEachAsync<T>(this IEnumerable<T> source, int dop, Func<T, Task> body)
```
Comment (martijn00) [inline Minio/Helper/Utils.cs:200]: Just take in the list of Tasks directly and depending on the .NET platform wrap it in a Task.WhenAll or Parallel.ForEachAsync

**Review by martijn00 — CHANGES_REQUESTED** [review]

File: Minio.Functional.Tests/FunctionalTest.cs [hunk 4]
```diff
@@ -332,13 +332,20 @@ internal static async Task RunCoreTests(MinioClient minioClient)
             GetBucketPolicy_Test1(minioClient)
         };
 
+#if NET6_0_OR_GREATER
```
Comment (martijn00) [inline Minio.Functional.Tests/FunctionalTest.cs:335]: Please move this to the helper method and do the switch per platform there.

File: Minio/Helper/Utils.cs [hunk 5]
```diff
@@ -196,6 +197,21 @@ internal static bool IsSupersetOf(IList<string> l1, IList<string> l2)
         return !l2.Except(l1).Any();
     }
 
+    public static Task ForEachAsync<T>(this IEnumerable<T> source, Func<T, Task> body)
+    {
+        var MaxNoOfParallelProcesses = 4;
```
Comment (martijn00) [inline Minio/Helper/Utils.cs:202]: `var maxNoOfParallelProcesses = 4;` non capital since it is a local var

**Review by martijn00 — CHANGES_REQUESTED** [review]

File: Minio/Helper/Utils.cs (hunk 5 above)
Comment (ebozduman) [reply Minio/Helper/Utils.cs]: Done

File: Minio.Functional.Tests/FunctionalTest.cs (hunk 4 above)
Comment (ebozduman) [reply Minio.Functional.Tests/FunctionalTest.cs]: Done

**Review by martijn00 — APPROVED** [review]

**Comment by ebozduman** [comment]
@martijn00 
Could you please review the latest one more time.
Thanks

**Review by martijn00 — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**